From patchwork Wed Feb 26 07:44:21 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405443
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 5E9A014BC
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:44:57 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 352F221D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:44:57 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="Mo7o+9Y/"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727278AbgBZHoy (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:44:54 -0500
Received: from mail-pg1-f202.google.com ([209.85.215.202]:48905 "EHLO
        mail-pg1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726587AbgBZHoy (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:44:54 -0500
Received: by mail-pg1-f202.google.com with SMTP id h14so1288318pgd.15
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:44:52 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=6pHSqsqhtQpsZBK8Kdeyx8XvutwVtDm2Afb6gQYh6DM=;
        b=Mo7o+9Y/ti+/X/PgbqFLsbuMoAos1ylIUguffoKM/q1+tZMkkLiS3bKoF1Xt71eQMO
         SwH73Tf5agOP65C8J1jNOrAFkhHypCfXDmdtYlFy1SmGljB2ueHbz+GTszrV+ifKTkvm
         WcQAniJR0v19zRpV2RgFwQOZRqUonvbBc5l2lVK8+Zj7JeYSE8H4zos8gZWMattSJ7fK
         SGCF7Vg24sDbLa5RTqjLOJF4+PnKSpFR4GKiHJv+clh60Gl1qJjdCLwb5MipvQTYM3NZ
         VtakBEUOoGrQkTI98Ope7F59FyS9nhrfw1d2+jPL17TER4LUQedF786FMWAzSr7MWk27
         Yg7A==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=6pHSqsqhtQpsZBK8Kdeyx8XvutwVtDm2Afb6gQYh6DM=;
        b=pO1oCBQGZcdVC1lDf4aY2gaa8CnfqD56DopfNLwWd6/t3wT6rfpdmYUFQf/somqBWd
         00zOf6/O6DCxF5gA33c8nnSWxCnCvd4mzXVvIASfrb6AcnIn8oS9vt8dlc4jCUBLjSV4
         cPSsE9H/qWdKj/NK50J2TTtDfj6ThORJZbk4WywZcosyRLtXZGwFC7Su7ilhpD6qoUYq
         Tln8LDkuNqL3b1xrI9K9JafrocUc/pSNCFaAmKUSiVSiSkfVWQ4x6xdw5DEm1ySmIa0i
         m/Dc35KMAQDA88lrNeBKwPHjFnOkgQ8pfNhQMqwgy2Rn1J0vAhqwdqEMvQnE0dMPkz/4
         cXuQ==
X-Gm-Message-State: APjAAAVMJ5RFt6FNZBCs1IXCsxnioim2EY4jjaala1mi1R4O7U3puRrE
        uahij6XaARoZ8e2JI1DI+4H/sJIJdyejOEDwGAqpaW6qanjeerBlK301/itOPGYKq6yzM0AnECA
        U8kxRzBI403lUp1EDkP5466r1AkwS8CUCS3hEk0t5vUBYzZzUF/M0+Q==
X-Google-Smtp-Source: 
 APXvYqwaNVMvUNxZClWOF2TGyEiGmrzZMxTbrVTH2vXi7Z4vUDkf+R75cl/YWQFkXRVTZnDSna6Sg7oraQ==
X-Received: by 2002:a63:6d01:: with SMTP id i1mr2528728pgc.55.1582703091894;
 Tue, 25 Feb 2020 23:44:51 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:21 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-2-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 1/7] x86: emulator: use "SSE2" for the target
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

The movdqu and movapd instructions are SSE2 instructions. Clang
interprets the __attribute__((target("sse"))) as allowing SSE only
instructions. Using SSE2 instructions cause an error.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/emulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/emulator.c b/x86/emulator.c
index 8fe03b8..2990550 100644
--- a/x86/emulator.c
+++ b/x86/emulator.c
@@ -658,7 +658,7 @@ static bool sseeq(sse_union *v1, sse_union *v2)
     return ok;
 }
 
-static __attribute__((target("sse"))) void test_sse(sse_union *mem)
+static __attribute__((target("sse2"))) void test_sse(sse_union *mem)
 {
     sse_union v;
 

From patchwork Wed Feb 26 07:44:22 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405445
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 87A78924
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:44:57 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 67EC221D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:44:57 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="NNMxzPyW"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727304AbgBZHo4 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:44:56 -0500
Received: from mail-yw1-f73.google.com ([209.85.161.73]:49299 "EHLO
        mail-yw1-f73.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726587AbgBZHoz (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:44:55 -0500
Received: by mail-yw1-f73.google.com with SMTP id j63so2654918ywf.16
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:44:55 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=MPs9SjjXPdQ3g+8JCLKgeYABN+oF1DJ3MQKhqKdCCeE=;
        b=NNMxzPyWcPdUrihQXaU4e+ps8WMo5ta29zZkt36qf+rJEnOP0Kw6dqoKJdJhk0aygv
         oTE19LlIqgZ2l0Vq8k6FLKiIKu/pdbAjtn9/x96jKL+vI3ucym0NKLv8QcsmVS1Q0AaW
         6kLXqu0HuzSotIx4atbnBPCDX2UwRZvPwmWNCS59YbCNbDT1z5xgYkj9yi1piVa2dZZe
         JX+xuSNPPf/LBvhwqHa+4aJQSk9ISMU2e5dGMW6tLudalwlVtwlHAuXYQIgd3YKrPQVx
         55Vr2Q6MNydyT3hbpHyXClNigMJoGkNXhRv+1P9W1yyxyKS4mOEblvxAKGzZQfuZUpOR
         OBRg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=MPs9SjjXPdQ3g+8JCLKgeYABN+oF1DJ3MQKhqKdCCeE=;
        b=MuMBP6aYIWLOy9vEBQaJj/3EllDFPMGDISbLkAiSG8hp37TSWfUwIIWUL07AD40s7/
         +YRXikVMgSDGkwTUL9R4pMkZMvsoabF9rZ6z+Bd9JXbOaXopRpLHRKqDXE6OZ9+4gruR
         ndG9558af0Uiv4LsayOpueymmBsnrU5G32v8Jqin9JxooMYh7dUaQRnlsU+mBsiX/CsJ
         wkvWRTx0BjElLrUutw4VTrdoMCwAN+U9voX2GHvYE7yxK4Wba1VWdZJeR49QRsRstOtT
         GscD1ZPlDmHMXr8v1fNk5E9WTl2kFpuX7Q2JXdENOYo29rLpepmoYbaEzPqYiNdx8l6F
         OGRQ==
X-Gm-Message-State: APjAAAWj5FFbU2/k3/D/o0rmCM8j0MqKPMaet/iSZdOcMjqXQ602HAFx
        JlHxC3T/fXjnpeZNSLGBFx1/24e6EXiv2b+UqdTr6/x6Z19WDSMM6EvpBA6U2C5PFQa/xKQfybb
        togImUlnEMvfij6mZOJ7f1FVj9QMIStdsVqiBhRl7iF+2mJqwc0Vi5w==
X-Google-Smtp-Source: 
 APXvYqxw8sbZEcncXybGIuNwvlSVTwa77cerJtDcBY66xCmQpF7kxrAaCfi2ceaI+b4Kiee8k96iOxaT3A==
X-Received: by 2002:a81:1bc6:: with SMTP id
 b189mr2351619ywb.275.1582703094538;
 Tue, 25 Feb 2020 23:44:54 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:22 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-3-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 2/7] pci: use uint32_t for unsigned long values
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

The "pci_bar_*" functions use 64-bit masks, but the results are assigned
to 32-bit variables. Use 32-bit masks, since we're interested only in
the least significant 4-bits.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 lib/linux/pci_regs.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/linux/pci_regs.h b/lib/linux/pci_regs.h
index 1becea8..3bc2b92 100644
--- a/lib/linux/pci_regs.h
+++ b/lib/linux/pci_regs.h
@@ -96,8 +96,8 @@
 #define  PCI_BASE_ADDRESS_MEM_TYPE_1M	0x02	/* Below 1M [obsolete] */
 #define  PCI_BASE_ADDRESS_MEM_TYPE_64	0x04	/* 64 bit address */
 #define  PCI_BASE_ADDRESS_MEM_PREFETCH	0x08	/* prefetchable? */
-#define  PCI_BASE_ADDRESS_MEM_MASK	(~0x0fUL)
-#define  PCI_BASE_ADDRESS_IO_MASK	(~0x03UL)
+#define  PCI_BASE_ADDRESS_MEM_MASK	(~0x0fU)
+#define  PCI_BASE_ADDRESS_IO_MASK	(~0x03U)
 /* bit 1 is reserved if address_space = 1 */
 
 /* Header type 0 (normal devices) */

From patchwork Wed Feb 26 07:44:23 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405447
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 0965C92A
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:04 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id DCF3621D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:03 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="cIZI/Ozm"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727312AbgBZHo7 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:44:59 -0500
Received: from mail-pl1-f201.google.com ([209.85.214.201]:36871 "EHLO
        mail-pl1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727309AbgBZHo6 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:44:58 -0500
Received: by mail-pl1-f201.google.com with SMTP id t12so1409182plo.4
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:44:58 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=v0kAimqMmAthsdANDZDbLgds4YdFM+K+E13QIetrdbM=;
        b=cIZI/OzmBenIvKEfOgu5v9GuZKjhj0W7tXibySmLIK0kxwNxmm11dDO5LZ+zoSNCNC
         koLMIthzvo6MrgoGc1Sj5pQabViVGdweIhohtdgd+u/TAzv4hMsSeUWf9zEJ5OZBdLLf
         thdQpoHNT7Nqctpr8ECTxwD10RnSp6DJiUs51mWrpHDggnrkmNW7VX1UN3XRn/L3rHBN
         6fuTb/TjtpUrEwWHkvm0x/DYTYNZf9V1XkaEucYuz3sl1YLEVPfAkvuEq/V5uMNQNqUy
         OLS9Gl28rNFi02AHCd2sa81MzcIO+gqhDoDg8FjbPZhS3vY5RTccrP0VmDikwW4U3sdG
         MGWw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=v0kAimqMmAthsdANDZDbLgds4YdFM+K+E13QIetrdbM=;
        b=NmYe5NmbSC9bf+qiUGbBdiRquGhECeDzqWi+mz/GSHVrfwh7Hw3LgCIgduGWL6Gol+
         ZBIxNHrK9oa9iE0zOr4a8mzTjKltk9T36dlHryEvjYhA1Y+n0FUvz2kPYm9MygUWybJv
         XBykPm8y/gMDU/RsfDpR69BiURq0LkpD9RdB5xaQnsh0sXb9fNJx5shLY2egjVBYxdMu
         dgiFGKUKqsYuY5AYtz7jffXi64ZMl+S7xKiB+aIkGMRkMOXSXCgVfoAfxJoQ4NlaSLSs
         o7uTd6gs8asilJdTG32gE3KKsAEssqQeXXqzR9JJH8zMd2/6w1Rz0i5Fmgo3Ds6TSoiU
         KgMw==
X-Gm-Message-State: APjAAAWxpF6XED1rqAiz/4oXL0vm+PnayvqgFi2jupnOKeppaHG3h2sv
        wUMw52vMw26E8xqMNEg/CjJbRxkYODf3kOprT6vAnRxUDXl4j7j7IIxtBXvT4CYdhaOPNjuZrj0
        cqyjO6pMGeqYKASdUXtYgz34Tx3qhxGswebDaDJZbDdzl0b5iHKlu1Q==
X-Google-Smtp-Source: 
 APXvYqzy8ZpIwcAB1hYmYxTBKjYDuBodQVRA8fiBO589/S5pjHtr8WOVH6+TpF15OaNl8N2F262Caw79TQ==
X-Received: by 2002:a63:4b07:: with SMTP id y7mr2528703pga.272.1582703096932;
 Tue, 25 Feb 2020 23:44:56 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:23 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-4-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 3/7] x86: realmode: syscall: add explicit size
 suffix to ambiguous instructions
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

Clang requires explicit size suffixes for potentially ambiguous
instructions:

x86/realmode.c:1647:2: error: ambiguous instructions require an explicit suffix (could be 'cmpb', 'cmpw', or 'cmpl')
        MK_INSN_PERF(perf_memory_load, "cmp $0, (%edi)");
        ^
x86/realmode.c:1591:10: note: expanded from macro 'MK_INSN_PERF'
                      "1:" insn "\n"                            \
                       ^
<inline asm>:8:3: note: instantiated into assembly here
1:cmp $0, (%edi)
  ^

The 'w' and 'l' suffixes generate code that's identical to the gcc
version without them.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/realmode.c | 6 +++---
 x86/syscall.c  | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/x86/realmode.c b/x86/realmode.c
index f5967ef..31f84d0 100644
--- a/x86/realmode.c
+++ b/x86/realmode.c
@@ -1644,7 +1644,7 @@ static void test_perf_memory_load(void)
 {
 	u32 cyc, tmp;
 
-	MK_INSN_PERF(perf_memory_load, "cmp $0, (%edi)");
+	MK_INSN_PERF(perf_memory_load, "cmpw $0, (%edi)");
 
 	init_inregs(&(struct regs){ .edi = (u32)&tmp });
 
@@ -1657,7 +1657,7 @@ static void test_perf_memory_store(void)
 {
 	u32 cyc, tmp;
 
-	MK_INSN_PERF(perf_memory_store, "mov %ax, (%edi)");
+	MK_INSN_PERF(perf_memory_store, "movw %ax, (%edi)");
 
 	init_inregs(&(struct regs){ .edi = (u32)&tmp });
 
@@ -1670,7 +1670,7 @@ static void test_perf_memory_rmw(void)
 {
 	u32 cyc, tmp;
 
-	MK_INSN_PERF(perf_memory_rmw, "add $1, (%edi)");
+	MK_INSN_PERF(perf_memory_rmw, "addw $1, (%edi)");
 
 	init_inregs(&(struct regs){ .edi = (u32)&tmp });
 
diff --git a/x86/syscall.c b/x86/syscall.c
index b4f5ac0..b7e29d6 100644
--- a/x86/syscall.c
+++ b/x86/syscall.c
@@ -38,7 +38,7 @@ static void handle_db(struct ex_regs *regs)
 
 /* expects desired ring 3 flags in rax */
 asm("syscall32_target:\n"
-    "   cmp $0, code_segment_upon_db(%rip)\n"
+    "   cmpl $0, code_segment_upon_db(%rip)\n"
     "   jne back_to_test\n"
     "   mov %eax,%r11d\n"
     "   sysretl\n");

From patchwork Wed Feb 26 07:44:24 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405449
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 31E271395
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:04 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 1257821D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:04 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="PXhC3lkx"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727334AbgBZHpB (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:45:01 -0500
Received: from mail-pf1-f202.google.com ([209.85.210.202]:49759 "EHLO
        mail-pf1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727307AbgBZHpB (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:45:01 -0500
Received: by mail-pf1-f202.google.com with SMTP id z17so1494518pfq.16
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:45:00 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=E11peBCBeUkU5QqrM9F+hDGSoXytqmeuQIcQ91AaOag=;
        b=PXhC3lkx7kzJcBy6Q3o9LjR3kjP+Qb7O7T1Pjc6CF7JJl8lurJ2pRvfGajoRWtjV0B
         DLBkk9ql1m6PPU+KR7A5Gy1Vo0Tii9gD9esRBzP/G++Uv8qSs16kofc9FrfUDd3zYds0
         Kio9RC9B5YEEjcpdI6WGI2mBPxx7HWA53lu7JfeDVoiOk9yr5CadDP1TrfoxsWJrz9kV
         XiAR9BjpkaF/+Cz30vu0wqNphR1MY3nIxDBMPzOOLyzNhicHNnyccZxUQty8DtoJoTwl
         h70hV+LWAhoHCi9gj8tEqXcoXCL5IVUbccky7kbed9aGZ9/VXwFA+k3P+ull6JxS4nis
         +tPA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=E11peBCBeUkU5QqrM9F+hDGSoXytqmeuQIcQ91AaOag=;
        b=TZ7kKtU3scqr4jKnFenYviT88WkcqHhGnx6Kwan5j12pn+Bo+AqhPKL2JPLBi3fbGQ
         sEuOd0khD/GLpe3vVcnRSTvIKmEsprdCFLH3xGEP2AjWyQU2PuU6AYC/xxZHxR34l8DQ
         Eoyr0PIapXU8hxt9ctO/2Ph2DEjVTE06WAmQPSLtFHCrgGtl5c4JCtfFBc1JuDMgoH0t
         +CfI3KEaAFKYCjxE0uAMELdvUC5WiBlof/gdNqMYU7YfTvU0uLteir6gdAYBNwYDPomF
         65Mceyz7NLoVrNGAUyGHCVyq60pMlR04o/o1ox3VXdS3eMnSfVRG2xPn0Iq+a62LrgeA
         9VJQ==
X-Gm-Message-State: APjAAAUKU0+bGC5zzLl2L1DQyElZ3QGGSRozSGo24Ro5X/Bzl5SkbSYH
        mgWM970IPKIFy/LweCsZ9dMnjaC8UeTO8JqI96y5f5k+WfUQ6Y9XaKxcimnHx909zx0EI246Src
        RiuMg7BJVnAqlXoqHrg52vXUuCJhiuQHC54xXOvUiguzWcHoHCYNwJw==
X-Google-Smtp-Source: 
 APXvYqwnF4T9oiM0Mq/9sMFYg+ye9CPyJsFfHWxWUaYTEotetUTGSucmt3tgFaUqi3bXPtsizq+glA7jwQ==
X-Received: by 2002:a63:1d18:: with SMTP id d24mr2463870pgd.189.1582703099990;
 Tue, 25 Feb 2020 23:44:59 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:24 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-5-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 4/7] svm: change operand to output-only for
 matching constraint
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

According to GNU extended asm documentation, "the two operands [of
matching constraints] must include one input-only operand and one
output-only operand." So remove the read/write modifier from the output
constraint.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/svm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/x86/svm.c b/x86/svm.c
index aa3d995..ae85194 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -288,8 +288,8 @@ static void test_run(struct test *test, struct vmcb *vmcb)
             "cli \n\t"
             "stgi"
             : // inputs clobbered by the guest:
-	      "+D" (the_test),            // first argument register
-	      "+b" (the_vmcb)             // callee save register!
+	      "=D" (the_test),            // first argument register
+	      "=b" (the_vmcb)             // callee save register!
             : [test] "0" (the_test),
 	      [vmcb_phys] "1"(the_vmcb),
 	      [PREPARE_GIF_CLEAR] "i" (offsetof(struct test, prepare_gif_clear))

From patchwork Wed Feb 26 07:44:25 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405451
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 2045D1395
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:06 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id F362E21D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:05 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="wPT135dK"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727335AbgBZHpE (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:45:04 -0500
Received: from mail-pl1-f201.google.com ([209.85.214.201]:47597 "EHLO
        mail-pl1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727346AbgBZHpD (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:45:03 -0500
Received: by mail-pl1-f201.google.com with SMTP id h3so1396412plt.14
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:45:03 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=yng0GPy0e4PuBKIocCGSBVAxQisSu+IFnd/3PrDUp3w=;
        b=wPT135dKnaFgTXeaqYhuIk8yZBRCIyobhOLCm8t1kPJ+3N6Mg7Pbl/ygvDL1ZEemoE
         hpeO/T8WFTnM9X7TSyMd2oes5fk3ExNY2FfP1f4SuE0B0O0bib5kARk88Zt8SyDY9F6A
         EMkkbrXM8V85v5mbMl/L4tQWgVJ7oa25qBZ7fNWgQtxwNgFn6+zooZt+KThVI2DorE1k
         H5clgDCFTd71rbXed+VdMoY/PIU3CA5b1oQA6ecSMSnDNteMFItESJff9kBhXSiYrtMG
         IGUay8QwnVwsNdnh6W1WKm/sn+hXUnzEffWU+oeMmfF17RN+q3a75+HJvJsMpGJOvy/q
         QwZQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=yng0GPy0e4PuBKIocCGSBVAxQisSu+IFnd/3PrDUp3w=;
        b=Hvk8odPs9kGGLMJWYrqJiM8kNacLMHfkfPsLnTvtd32EbWfE2IaoeQDyU3KznhMbDw
         fGJn68KdasD0Srb1Y7iq2uhwPi0nkhIK99q3InHsVA/gYztYfRXiuqXTjG7idvIgY8xs
         93De/HygEdae1MaVYtnRqJiBiRNr/6Jz/V97z0NDzNzGdRQBirUY00UuohwnUDyh+smB
         nmSjncMxfHWQyDxf+qFinLQhpY5SxCvK8qRGOLnQ/TAzPcht/94JvKFvJEoK9bC7eXMg
         rOtNK42rh62ScGxKJ6YgxYlYQGx64P3ikLfI7wKJYaRWCV8YuDCXp/qYQ++ySkVyw81t
         Z9dQ==
X-Gm-Message-State: APjAAAVN6rFiDf7uWE/VdlG1O2oUTfhmlnKHJBz/hRk4/SM1et2+SSZi
        DQKOKZy39dOTJUis3Z4566+yBhlm+StyvEQDO9LeoScCCD7aYMPC24SI088Gu18Yhp3Gx3g2Wzy
        KBoJs9b0+Wm4Nbi1jR3qqo2XKIwoDEL13f9gUJ+QyFQzvjj0ojzhIYQ==
X-Google-Smtp-Source: 
 APXvYqwO4Nifwise5A6AMPigFvw8PHLykHiLaZZV3Ll47ZZeOC2ZNA3DL1hkuIBhhtFSgn1rH1UPcR8LHw==
X-Received: by 2002:a63:f10c:: with SMTP id f12mr2665678pgi.386.1582703102494;
 Tue, 25 Feb 2020 23:45:02 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:25 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-6-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 5/7] svm: convert neg shift to unsigned shift
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

Shifting a negative signed value is undefined. Use a shift of an
unsigned value instead.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/svm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/svm.c b/x86/svm.c
index ae85194..17be4b0 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -1148,7 +1148,7 @@ static bool npt_rw_l1mmio_check(struct test *test)
 }
 
 #define TSC_ADJUST_VALUE    (1ll << 32)
-#define TSC_OFFSET_VALUE    (-1ll << 48)
+#define TSC_OFFSET_VALUE    (~0ull << 48)
 static bool ok;
 
 static void tsc_adjust_prepare(struct test *test)

From patchwork Wed Feb 26 07:44:26 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405453
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3FCBD1395
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:08 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 1FA7121D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:08 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="bv274DrR"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727359AbgBZHpH (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:45:07 -0500
Received: from mail-pg1-f202.google.com ([209.85.215.202]:38806 "EHLO
        mail-pg1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727350AbgBZHpG (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:45:06 -0500
Received: by mail-pg1-f202.google.com with SMTP id x16so1308867pgg.5
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:45:05 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=7wW5RNchVTlItAFZhajd4+Ejg/4+T3uP4HbdOkXgmio=;
        b=bv274DrRU5vCxbKG6elhIw5NMfIZycw7uUH9xLoFC/9gN49dxS10JixwugF/AJJNZ3
         Y6ukDxMbkkJJQpA0b65qeQ3YtVpxdvLtg/Cd68oQyeubXgSARMNow4TVOVGwt0QomycA
         2alw6ld2T+6SLgyTnYsrg3ffSBoFvRN5FRy2gptYk6htqdZx3j6wv81UH/f+u6a3ds7N
         pHb/1z4ueRV1ZrwV5QZAApWPYrxCgPuM0H//CVg3gBbh3Nr9GGlPRIfDoJCxpo7o4oAj
         FMJhmUc/yE/NAo2z9UEjiJGDloYwbvWtFerQ6RsaM0DrcRJVvkpiPv0mLwCSrjtley6R
         j0RA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=7wW5RNchVTlItAFZhajd4+Ejg/4+T3uP4HbdOkXgmio=;
        b=bzHdIOapezIhTwBYL46DTcf7fmywbMkw9DNTP/wqqhimZqZnVRij3YuZ3kouh1NRWI
         JdghrKBlVfGGLBN4nmyH4R5qEAbKeOOKp3eFVIsIVD3PzaBKvxfUnOmoWWctOyqGcNq1
         ZvMvck5mDXWmbvdi1W/Dw+71AnF+V9B8ppKVfYGA5oyXB33+YyZ8UE4lQAZPv81z0CS1
         3fzfvqq6ePyTl1z8x8zhcGNBEBv5hQpi5a/UBf7BIxKIyvsZaiRNY8TzZKMjjuhkadSM
         ba0k3I+DmtPYtskkwBH1+k/bIkOL6eRRR4id4lH1GUzHXxZ6d2YZLYpWsti8eB85zTvv
         C6qQ==
X-Gm-Message-State: APjAAAX09i86D1rO4gS4GoCFRFIBjCApW3xk44VbZkoa7UhkZ/stYmc6
        ztDNb0Pcu5yBYChL4moWILxp4YL+zbccb3Duq5vkU4M+tU82p1r/wtOqV+hLJ+ewfMDZkg0DdQ/
        maaWu6T/BXxRMMooDuKarg5KYvvfwdgn8nd8WRUfXB6ATpfufuMluwA==
X-Google-Smtp-Source: 
 APXvYqx4XzVQ6tdvnt+VZ1Zz7gfRbv2eBMmPBXI/kfHLe08x8HtKpaYvBPgUlaHNp5CFOvsOCgqWiIVZnA==
X-Received: by 2002:a63:7e49:: with SMTP id o9mr2670979pgn.80.1582703105074;
 Tue, 25 Feb 2020 23:45:05 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:26 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-7-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 6/7] x86: VMX: use inline asm to get stack
 pointer
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

The only supported use of a local register variable is to specify
registers for input and output operands when calling Extended asm. Using
it to automatically collect the value in the register isn't supported as
the contents of the register aren't guaranteed. Instead use inline asm
to get the stack pointer explicitly.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/vmx_tests.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index a7abd63..ad8c002 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -2165,8 +2165,9 @@ static void into_guest_main(void)
 		.offset = (uintptr_t)&&into,
 		.selector = KERNEL_CS32,
 	};
-	register uintptr_t rsp asm("rsp");
+	uintptr_t rsp = 0;
 
+	asm volatile ("mov %%rsp, %0" : "=r" (rsp));
 	if (fp.offset != (uintptr_t)&&into) {
 		printf("Code address too high.\n");
 		return;
@@ -3261,8 +3262,9 @@ static void try_compat_invvpid(void *unused)
 		.offset = (uintptr_t)&&invvpid,
 		.selector = KERNEL_CS32,
 	};
-	register uintptr_t rsp asm("rsp");
+	register uintptr_t rsp = 0;
 
+	asm volatile ("mov %%rsp, %0" : "=r" (rsp));
 	TEST_ASSERT_MSG(fp.offset == (uintptr_t)&&invvpid,
 			"Code address too high.");
 	TEST_ASSERT_MSG(rsp == (u32)rsp, "Stack address too high.");

From patchwork Wed Feb 26 07:44:27 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11405455
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A7C4092A
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:10 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 86E1921D7E
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 07:45:10 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="YC1fbhcW"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727392AbgBZHpJ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 02:45:09 -0500
Received: from mail-pj1-f73.google.com ([209.85.216.73]:32953 "EHLO
        mail-pj1-f73.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727368AbgBZHpI (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 02:45:08 -0500
Received: by mail-pj1-f73.google.com with SMTP id d22so3007060pjz.0
        for <kvm@vger.kernel.org>; Tue, 25 Feb 2020 23:45:08 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=8HBWKlQYG3cu24wlZPhv8FmjF0s+tjjwCin6uZ9bx5M=;
        b=YC1fbhcW5B+/Tqz5q1XH59Ir37woGfBwArFpTS+g300ghGdVAeFy+s9jfd8kdIxPMz
         7S3owuhX5rGdNPUPdxMAP0N9EIlMc62d7NoCcFLRm4U6jqSlaAYX0DT/xX96fLbnYE/V
         POscfmeB/sh+6sjcJhE4s8byPNm0FrwuFNiWTuEjFZ0K38+Xrh4/7SZtYuNGTxK2v334
         wdgrNXW2AkQ+1RTJj9uhJLM2kqI8duz5MxN0jGkz33HTwAopK3AMux4ftD8gCoo5YZrx
         xzxP8OXCsNoTJpq0OCcZqGUujaUboM69iTAtZWjErMJM3XZ/a40yWyXrcskGUvzgh7+c
         ZPdQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=8HBWKlQYG3cu24wlZPhv8FmjF0s+tjjwCin6uZ9bx5M=;
        b=HOtClCnS9+Z+WUHwMFST3kIQLL0eFzJ5NrSbWA7WV+gTIw8IDgfnp1rl9tt9kmKYQf
         uf1JhOsE8VyNvajrVcjsUtaONB9dUplYRlYlTvPzB6qVWSofl82fGLopjkZgY3Q7SPzC
         1HXxGZnOqwMYdyXGlX5iqH0ZTlYClf1KvP4Su6mB5N6tuQP0UM1WzkHEgTrd2PpOXL4f
         smVFoOXwIRSonslfiS7iNCOVkkjo18zAlwezJPHyQkDYv5c2psdxRqYqVLeNtMCKAfoD
         v0JgM7dFPDptOw+Vh44V2ksUlwVDtqRgtCUNPpUk+FYBFLsv5VdRcPCpBPv39zAfuC/I
         bZLg==
X-Gm-Message-State: APjAAAU2e37SWGqsXd5pPn7JPN6CW87g5iAXX7PBQzOu2U8qEOQ/s39Q
        C6uZogZ60NjtAO6r/Hq7WVSYhdx9Sgtsb/IVUkgNufVWlaCbVLqwHWcjk/EV0Ly3HnesNiI11jI
        wxSAB1l+0m5CwZD252wtqqLJnl1Jqlsoh6cbK4OKHKcnSHwlKjdSFjA==
X-Google-Smtp-Source: 
 APXvYqyKuoqOSEuCd1dMT5Gj0tPCF63SaV/rlATHpwg4BViQ/y8XcZFUrvhJ4lA3WRQRnqxgN0NUUm9jSQ==
X-Received: by 2002:a63:f20d:: with SMTP id v13mr2552778pgh.34.1582703107715;
 Tue, 25 Feb 2020 23:45:07 -0800 (PST)
Date: Tue, 25 Feb 2020 23:44:27 -0800
In-Reply-To: <20200226074427.169684-1-morbo@google.com>
Message-Id: <20200226074427.169684-8-morbo@google.com>
Mime-Version: 1.0
References: <20200226074427.169684-1-morbo@google.com>
X-Mailer: git-send-email 2.25.0.265.gbab2e86ba0-goog
Subject: [kvm-unit-tests PATCH 7/7] x86: VMX: the "noclone" attribute is
 gcc-specific
From: morbo@google.com
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Bill Wendling <morbo@google.com>

Don't use the "noclone" attribute for clang as it's not supported.

Signed-off-by: Bill Wendling <morbo@google.com>
Signed-off-by: Oliver Upton <oupton@google.com>
---
 x86/vmx_tests.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index ad8c002..ec88016 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -4976,7 +4976,10 @@ extern unsigned char test_mtf1;
 extern unsigned char test_mtf2;
 extern unsigned char test_mtf3;
 
-__attribute__((noclone)) static void test_mtf_guest(void)
+#ifndef __clang__
+__attribute__((noclone))
+#endif
+static void test_mtf_guest(void)
 {
 	asm ("vmcall;\n\t"
 	     "out %al, $0x80;\n\t"
